upgrades/upgrade_testing_framework/clients/rest_client_default.py

```diff
@@ -0,0 +1,105 @@
+        params = {"v": "true", "pretty": "true"}
+        return ops.perform_get(rest_path = rest_path, params = params)
+
+    def get_doc_by_id(self, port: int, index: str, doc_id: str) -> ops.RESTResponse:
+        """
+        Get a document by its ID
+        """
+        rest_path = ops.RESTPath(port = port, suffix = f"{index}/_doc/{doc_id}")
+        params = {"pretty": "true"}
+        return ops.perform_get(rest_path = rest_path, params = params)
+
+    def post_doc_to_index(self, port: int, index: str, doc: dict) -> ops.RESTResponse:
+        """
+        Post a single document to an index
+        """
+        rest_path = ops.RESTPath(port = port, suffix = f"{index}/_doc")
+        params = {"pretty": "true"}
+        headers={"Content-Type": "application/json"}
+
+        return ops.perform_post(rest_path = rest_path, data = json.dumps(doc), headers = headers, params = params)
+
+    def create_snapshot(self, port: int, repo: str, snapshot_id: str) -> ops.RESTResponse:
+        """
+        Create a snapshot of the cluster into a repo
+        """
+        rest_path = ops.RESTPath(port = port, suffix = f"_snapshot/{repo}/{snapshot_id}")
+        params = {"pretty": "true", "wait_for_completion": "true"}
+
+        return ops.perform_post(rest_path = rest_path, params = params)
+
+    def get_snapshot_by_id(self, port: int, repo: str, snapshot_id: str) -> ops.RESTResponse:
+        """
+        Get a snapshot by its ID
+        """
+        rest_path = ops.RESTPath(port = port, suffix = f"_snapshot/{repo}/{snapshot_id}")
+        params = {"pretty": "true"}
+        return ops.perform_get(rest_path = rest_path, params = params)    
+
+    def get_snapshots_all(self, port: int, repo: str) -> ops.RESTResponse:
+        """
+        Get all snapshots from a repo
+        """
+        rest_path = ops.RESTPath(port = port, suffix = f"_snapshot/{repo}/_all")
+        params = {"pretty": "true"}
+        return ops.perform_get(rest_path = rest_path, params = params)
+    
+    def register_snapshot_dir(self, port: int, repo: str, dir_path: str) -> ops.RESTResponse:
+        """
+        Register a directory on the node as the place to store snapshots for an snapshot repo
+        """
+        rest_path = ops.RESTPath(port = port, suffix = f"_snapshot/{repo}")
+        headers={"Content-Type": "application/json"}
+
+        register_args = {
+            "type": "fs",
+                "settings": {
+                    "location": dir_path
+            }
+        }
+
+        return ops.perform_post(rest_path = rest_path, data = json.dumps(register_args), headers = headers)
+
+    def restore_snapshot(self, port: int, repo: str, snapshot_id: int) -> ops.RESTResponse:
+        """
+        Restore a snapshot
+        """
+        rest_path = ops.RESTPath(port = port, suffix = f"_snapshot/{repo}/{snapshot_id}/_restore")
+        params = {"pretty": "true"}
+        return ops.perform_post(rest_path = rest_path, params = params)    
+
+
```

Comment:
Minor thing: but lots of extra lines down here and missing newlines between functions. I know we're not in full compliance with the linting yet, but can you check that you're not adding more issues in the new files?

Reply:
Sure thing!  We're gonna need to do a major linting pass next year.

Reply:
Fixed in next rev